lib/Partitioner/Partitioner.cpp

```diff
@@ -35,6 +42,65 @@ static uint64_t updateUsedMem(const std::set<Storage *> &usedStorage,
   return ret;
 }
 
+void Partitioner::dumpDAG(llvm::StringRef dotFilename) {
+  if (partitions_.size() == 0)
+    return;
+  std::map<BackendKind, const char *> backendName = {
+      {BackendKind::Interpreter, "Interpreter"},
+      {BackendKind::OpenCL, "OpenCL"},
+      {BackendKind::CPU, "CPU"},
+      {BackendKind::Habana, "Habana"}};
+  std::map<BackendKind, const char *> backendColor = {
```

Comment:
Instead of mapping explicitly each Backend to colors, which means we must update this when we add a new Backend, you could do what we do when assigning colors to Nodes in `dumpNode()` in `Graph.cpp` -- create a large array of `colorNames` and then use `llvm::hash_value` to get an index into it for the Backend. Actually, this code could probably be moved from `dumpNode()` to a helper that takes a string and returns a color string, and then shared in both places.